recipes/pyston_lite/meta.yaml

```
@@ -0,0 +1,93 @@
+{% set name = "pyston_lite" %}
+{% set pyston_version = "2.3.4" %}
+
+package:
+  name: {{ name|lower }}-split
+  version: {{ pyston_version }}
+
+source:
+  - url: https://github.com/pyston/pyston/archive/pyston_{{ pyston_version }}.tar.gz
+    sha256: 9989ce0f3a83611b11f79121d29ceec84c3cb4b4a8c3de4f8d97eb42042f4cee
+  # pyston-lite uses a preprocessing script from inside the LuaJIT repository (that is not
+  # packaged as part of the luajit package), but only as a build tool (ie is not included
+  # in the built artifacts).
+  - url: https://github.com/LuaJIT/LuaJIT/archive/e2c312e0deb874aa5fa8ce502c08d87deb38e82f.tar.gz
+    sha256: bff4764c19466def90667d2f7fcbd0781b8e62476b12105868f8e44babd03f8e
+    folder: pyston/LuaJIT/
+
+build:
+  number: 0
+
+  skip: True  # [not linux]
+  skip: True  # [py!=38]
+
+# In theory the requirements can/should be specified per-output, but in practice
+# conda-build will produce a corrupt package if the compiler dependency is not
+# additionally specified at this top scope. It is required to declare the
+# compiler dependency both here and in the outputs: section as well; removing
+# either will cause the c compiler to not be installed.
+#
+# It's also required to specify a "python" dependency in this section so that
+# conda-build knows that this package is sensitive to the python version.
+# (It does not infer this from the py!=38 jinja selector)
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+
+outputs:
+  - name: pyston_lite
+    # This 'script' line must go under 'outputs', not 'build'
+    script: build_pyston_lite.sh
+
+    # An "build" section is required, or else it seems to skip
+    # the build in certain build scenarios:
+    build:
+      # This build section cannot be empty or else it will crash the linter
+      script: true
+
+    requirements:
+      build:
+        # (See comment for the top-level requirements block)
+        - {{ compiler('c') }}
+        # NB: listing a "python" dependence in this section will cause the package to misbuild
+      host:
+        - python 3.8.* *_cpython
```

This will be automatically set by the above skips. Better omit this implementation detail of naming the python implementation.
```suggestion
        - python
```

---

I just tried this, and `build-locally.py` failed because it tried to build a 3.10 version of this package. I might be able to remove this pin by duplicating the skip line into this output, if you prefer that approach.

Also pyston-lite is by nature highly coupled with the CPython implementation, using many internal implementation details that aren't part of the C API. I added this pin because it doesn't work with Pyston, which is otherwise a highly-compatible implementation. I added a comment, hopefully this is ok.

---

Can confirm that skip statement does not prevent conda-build from trying to use python=3.10.